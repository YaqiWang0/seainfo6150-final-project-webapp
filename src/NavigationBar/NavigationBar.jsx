import React, {useState} from "react"
import SearchBar from "./SearchBar/SearchBar";
import UserButton from "./UserButton/UserButton";
import style from "./NavigationBar.module.css"
import { useHistory, useLocation } from "react-router-dom";
import LoginPage from "../LoginAndSignUp/LoginPage";
import {FirebaseContext} from "../Firebase";

const NavigationBar  = (props) => {
    //TODO: add login
    let history = useHistory();
    let location = useLocation();
    const [loggedIn, setLoggedIn] = useState(props.login.login);
    const [loginCredential, setLoginCredential] = useState({
                                                                       email: props.login.email,
                                                                       userId: props.login.userId,
                                                                     });

    const handleLogOut  = (firebase) => {
        firebase.doSignOut();
        setLoggedIn(false)
        setLoginCredential({
            userId: "",
            email: ""
        })
        if(location.pathname !== "/") {
            history.push(
                "/",
                {login: false}
            )
        }
    }

    const onLogin = () => {
        history.push(
            "/login_sign_up"
        )
    }

    const getUser = () => {
        history.push(
            "/user",
            {login: {
                login: loggedIn,
                email: loginCredential.email,
                userId: loginCredential.userId
                }}
        )
    }
    const userText = "USER"
    const loginText = "LOGIN"
    const logoutText = "LOGOUT"



    return (
        <div className={style.container}>
            <SearchBar login = {loggedIn} credential = {loginCredential}/>
            <div className={style.buttons}>
            <div className = {loggedIn == true ? style.hide : style.show}>
                <UserButton text = {loginText} onClick = {onLogin}/>
            </div>
            <div className = {loggedIn == true ? style.show : style.hide}>
                <UserButton text = {userText} onClick = {getUser}/>
            </div>
            <div className = {loggedIn == true ? style.show : style.hide}>
                <FirebaseContext.Consumer>
                    {firebase =>
                        <UserButton text = {logoutText} onClick = {() => handleLogOut(firebase)}/>
                    }
                </FirebaseContext.Consumer>
            </div>
            </div>
        </div>
    )
}

export default NavigationBar

