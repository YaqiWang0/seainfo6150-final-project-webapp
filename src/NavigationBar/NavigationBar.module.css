.container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    margin: 0;
    height: 5%;
    padding-right: 2em;
    background-image: url('../pictures/Black-Background-Picture.png');
}


.hide {
    display: none;
}

.show {
    display: block;
}

.func {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    align-items: center;
}

@media screen and (min-width: 0){
    .buttons {
        display: flex;
        flex-direction: column;
        padding-left: 2em;
    }
}

@media screen and (min-width: 700px){
    .buttons {
        display: flex;
        flex-direction: row;
        padding-left: 2em;
    }
}