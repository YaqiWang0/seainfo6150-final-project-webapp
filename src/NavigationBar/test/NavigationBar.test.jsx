import React from "react";
import NavigationBar from "../NavigationBar";

describe("NavigationBar test", () => {

    it('render successfully when login true', function () {
        const login = {
            login: true,
            userId: "NOFjyl43m1Xy3rB4umSRERq8M2z1",
            email: "123456@test.com"
        }
        const {container} = render(
            <NavigationBar login={login}/>
        );
        expect(container).toMatchSnapshot();
    })

    it('render successfully when login false', function () {
        const login = {
            login: false,
            userId: "NOFjyl43m1Xy3rB4umSRERq8M2z1",
            email: "123456@test.com"
        }
        const {container} = render(
            <NavigationBar login={login}/>
        );
        expect(container).toMatchSnapshot();
    })
})