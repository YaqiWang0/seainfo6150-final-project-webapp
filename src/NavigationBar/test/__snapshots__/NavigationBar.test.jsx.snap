// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer test render successfully when login false 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <div
        class="pic"
      />
    </div>
    <div
      class="func"
    >
      <div>
        <form
          action=""
          class="searchBar"
        >
          <input
            name="search"
            pattern=".*\\\\S.*"
            required=""
            type="search"
          />
          <button
            class="searchBtn"
            type="submit"
          >
            <span>
              Search
            </span>
          </button>
        </form>
      </div>
      <div
        class="buttons"
      >
        <div
          class="show"
        >
          <nav>
            <ul>
              <li>
                LOGIN
                <span />
                <span />
                <span />
                <span />
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="hide"
        >
          <nav>
            <ul>
              <li>
                USER
                <span />
                <span />
                <span />
                <span />
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="hide"
        >
          <nav>
            <ul>
              <li>
                LOGOUT
                <span />
                <span />
                <span />
                <span />
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Footer test render successfully when login true 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <div
        class="pic"
      />
    </div>
    <div
      class="func"
    >
      <div>
        <form
          action=""
          class="searchBar"
        >
          <input
            name="search"
            pattern=".*\\\\S.*"
            required=""
            type="search"
          />
          <button
            class="searchBtn"
            type="submit"
          >
            <span>
              Search
            </span>
          </button>
        </form>
      </div>
      <div
        class="buttons"
      >
        <div
          class="hide"
        >
          <nav>
            <ul>
              <li>
                LOGIN
                <span />
                <span />
                <span />
                <span />
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="show"
        >
          <nav>
            <ul>
              <li>
                USER
                <span />
                <span />
                <span />
                <span />
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="show"
        >
          <nav>
            <ul>
              <li>
                LOGOUT
                <span />
                <span />
                <span />
                <span />
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NavigationBar test render successfully when login false 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <div
        class="pic"
      />
    </div>
    <div
      class="func"
    >
      <div>
        <form
          action=""
          class="searchBar"
        >
          <input
            name="search"
            pattern=".*\\\\S.*"
            required=""
            type="search"
          />
          <button
            class="searchBtn"
            type="submit"
          >
            <span>
              Search
            </span>
          </button>
        </form>
      </div>
      <div
        class="buttons"
      >
        <div
          class="show"
        >
          <nav>
            <ul>
              <li>
                LOGIN
                <span />
                <span />
                <span />
                <span />
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="hide"
        >
          <nav>
            <ul>
              <li>
                USER
                <span />
                <span />
                <span />
                <span />
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="hide"
        >
          <nav>
            <ul>
              <li>
                LOGOUT
                <span />
                <span />
                <span />
                <span />
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NavigationBar test render successfully when login true 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <div
        class="pic"
      />
    </div>
    <div
      class="func"
    >
      <div>
        <form
          action=""
          class="searchBar"
        >
          <input
            name="search"
            pattern=".*\\\\S.*"
            required=""
            type="search"
          />
          <button
            class="searchBtn"
            type="submit"
          >
            <span>
              Search
            </span>
          </button>
        </form>
      </div>
      <div
        class="buttons"
      >
        <div
          class="hide"
        >
          <nav>
            <ul>
              <li>
                LOGIN
                <span />
                <span />
                <span />
                <span />
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="show"
        >
          <nav>
            <ul>
              <li>
                USER
                <span />
                <span />
                <span />
                <span />
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="show"
        >
          <nav>
            <ul>
              <li>
                LOGOUT
                <span />
                <span />
                <span />
                <span />
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </div>
</div>
`;
